core-bundle/src/Resources/contao/drivers/DC_Table.php

```diff
@@ -231,6 +232,58 @@ public function __construct($strTable, $arrModule=array())
 		}
 	}
 
+	/**
+	 * With this method, the ID of the current (parent) record can be
+	 * determined stateless based on the current request only.
+	 *
+	 * In older versions, Contao stored the ID of the current (parent) record
+	 * in the user session as "CURRENT_ID" to make it known on subsequent
+	 * requests. This was unreliable and caused several issues, like for
+	 * example if the user used multiple browser tabs at the same time.
+	 */
+	private function findCurrentPid(): ?int
+	{
+		if (!$this->ptable)
+		{
+			return null;
+		}
+
+		$id = ((int) Input::get('id')) ?: null;
+		$pid = ((int) Input::get('pid')) ?: null;
+		$act = Input::get('act');
+		$mode = Input::get('mode');
+
+		// For these actions the id parameter refers to the parent record
+		if (($act === 'paste' && $mode === 'create') || \in_array($act, array(null, 'select', 'editAll', 'overrideAll', 'deleteAll'), true))
+		{
+			return $id;
+		}
+
+		// For these actions the pid parameter refers to the insert position
+		if (\in_array($act, array('create', 'cut', 'copy', 'cutAll', 'copyAll'), true))
+		{
+			// Mode “paste into”
+			if ($mode === '2')
+			{
+				return $pid;
+			}
+
+			// Mode “paste after”
+			$id = $pid;
+		}
+
+		if (!$id || !$this->Database->fieldExists('pid', $this->strTable))
+		{
+			return null;
+		}
+
+		$objPid = $this->Database->prepare("SELECT pid FROM `$this->strTable` WHERE id=?")
```

Comment:
This should probably be updated to `$this->getCurrentRecord($id)` once https://github.com/contao/contao/pull/4770 is merged, so the record is already cached if used later 🙃

Reply:
See https://github.com/contao/contao/pull/4892.